src/main/java/faang/school/postservice/service/comment/CommentService.java

```diff
@@ -0,0 +1,57 @@
+package faang.school.postservice.service.comment;
+
+import faang.school.postservice.dto.comment.CommentDto;
+import faang.school.postservice.mapper.comment.CommentMapper;
+import faang.school.postservice.model.Comment;
+import faang.school.postservice.repository.CommentRepository;
+import faang.school.postservice.validator.comment.CommentValidator;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+    private final CommentRepository commentRepository;
+    private final CommentMapper commentMapper;
+    private final CommentValidator commentValidator;
+
+    @Transactional
+    public CommentDto createComment(Long postId, CommentDto commentDto) {
+        commentValidator.findPostById(postId);
+        Comment savedComment = commentRepository.save(commentMapper.toEntity(commentDto));
+        return commentMapper.toDto(savedComment);
+    }
+
+    @Transactional
+    public CommentDto updateComment(Long postId, Long commentId, CommentDto commentDto) {
+        commentValidator.findPostById(postId);
+        Comment comment = getValidComment(commentId);
+        commentValidator.checkUserRightsToChangeComment(comment, commentDto);
+        comment.setContent(commentDto.getContent());
+        return commentMapper.toDto(commentRepository.save(comment));
+    }
+
+    @Transactional(readOnly = true)
+    public List<CommentDto> getCommentsByPost(Long postId) {
+        commentValidator.findPostById(postId);
+        List<Comment> comments = commentRepository.findAllByPostIdSorted(postId);
+        return comments.stream()
+                .map(commentMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @Transactional
+    public void deleteComment(Long commentId) {
+        commentValidator.findCommentById(commentId);
+        commentRepository.deleteById(commentId);
+    }
+
+    private Comment getValidComment(Long commentId) {
+        commentValidator.findCommentById(commentId);
+        return commentRepository.findById(commentId).get();
```

Comment:
можем тут использовать Optional. на тот случай если метод вернет тебе нулл

Reply:
Метод не може вернуть null, потому что перед ним commentValidator. Тут вот у нас возник вопрос можно ли в валидатор например передать логику возвращения еще комментарий помимо проверки.
```suggestion
    public Comment getCommentById(Long commentId) {
        return commentRepository.findById(commentId).orElseThrow(() ->
                new RuntimeException("Comment not found!"));
    }
```

Но тут вопрос насколько уместно название getCommentById. Это бы сократило код, но возможно внесло бы путаницу и нарушило какие-то "принципы" написания кода

Reply:
тут я если что хочу заменить именно метод commentValidator.findCommentById в валидаторе

Reply:
Вообщем я убрал этот метод и добавил логику в валидатор которая возвращает комент. Не знаю на сколько это правильно но вроде сильно глаз не режет. Может можно еще добавить перегрузку чтобы был отдельно метод проверки и отдельно который еще и возвращает.

```
    public Comment findCommentById(Long commentId) {
        return commentRepository.findById(commentId).orElseThrow(() ->
                new CommentExceptionHandler("Comment not found!"));
    }
```

Вот что получилось, это метод из класса CommentValidator